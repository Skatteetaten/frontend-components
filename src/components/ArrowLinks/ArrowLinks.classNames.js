import { mergeStyleSets } from '@uifabric/merge-styles';
import { getTheme } from '@uifabric/styling';
import { MdIcons } from '../utils/icons';

export const getClassNames = () => {
  const { palette } = getTheme();
  const arrowIcon = "'" + MdIcons.icons.Back + "'";

  return mergeStyleSets({
    arrowLink: {
      display: 'block',
      position: 'relative',
      margin: '0 0 12px 26px',
      selectors: {
        '&::before': {
          fontFamily: MdIcons.fontFace.fontFamily,
          transform: 'rotate(180deg)',
          display: 'inline-block',
          position: 'absolute',
          content: arrowIcon,
          color: palette.skeColor.blue,
          lineHeight: 1,
          top: '9px',
          left: '-26px'
        },
        '&:hover::before': {
          left: -21,
          cursor: 'pointer'
        },
        '&:hover a': {
          textDecoration: 'underline',
          textDecorationColor: palette.skeColor.blue,
          color: palette.skeColor.darkBlue
        }
      }
    },
    arrowLinkA: {
      color: palette.skeColor.blue,
      textDecoration: 'none',
      fontWeight: 700,
      selectors: {
        '&:hover, &:active': {
          textDecoration: 'underline',
          textDecorationColor: palette.skeColor.blue,
          color: palette.skeColor.darkBlue
        }
      }
    }
  });
};
