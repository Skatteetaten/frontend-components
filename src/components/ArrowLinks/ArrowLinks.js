import * as React from 'react';
import classnames from 'classnames';
import { getClassNames } from './ArrowLinks.classNames';
import Step from '../StepList/Step';
import PropTypes from 'prop-types';

Step.propTypes = {
  /** Liste med linker **/
  links: PropTypes.arrayOf(PropTypes.object),
  /** Mulighet for å legge inn egen klasse for å overstyre stylingen */
  className: PropTypes.string
};

const ArrowLinks = props => {
  const [styles] = React.useState(getClassNames());
  return (
    <ul>
      {props.links.map((link, index) => (
        <li
          className={classnames(styles.arrowLink, props.className)}
          key={index}
        >
          <a href={link.path} className={classnames(styles.arrowLinkA)}>
            {link.text}
          </a>
        </li>
      ))}
    </ul>
  );
};

export default ArrowLinks;
