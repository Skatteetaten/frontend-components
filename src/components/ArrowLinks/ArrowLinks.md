** Link med pil **

```js
import ArrowLinks from '@skatteetaten/frontend-components/ArrowLinks';
const links = [
  {
    text: 'Dette er en link',
    path: '#'
  },
  {
    text: 'En annen  link',
    path: '#'
  }
];

<ArrowLinks links={links} />;
```
