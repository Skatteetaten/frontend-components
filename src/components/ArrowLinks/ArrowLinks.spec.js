import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import ArrowLinks from './ArrowLinks';

function oppsettMount() {
  return mount(
    <ArrowLinks links={[{ text: 'Dette er en link', path: '#' }]} />
  );
}

describe('Link komponent', () => {
  it('matcher snapshot', () => {
    const wrapper = oppsettMount();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
