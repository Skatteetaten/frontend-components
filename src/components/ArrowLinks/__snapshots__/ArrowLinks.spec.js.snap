// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link komponent matcher snapshot 1`] = `
<ArrowLinks
  links={
    Array [
      Object {
        "path": "#",
        "text": "Dette er en link",
      },
    ]
  }
>
  <ul>
    <li
      className="arrowLink-36"
      key="0"
    >
      <a
        className="arrowLinkA-37"
        href="#"
      >
        Dette er en link
      </a>
    </li>
  </ul>
</ArrowLinks>
`;
